src/test/java/uk/gov/justice/digital/zone/CuratedZoneTest.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.zone;
+
+import lombok.val;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.digital.config.BaseSparkTest;
+import uk.gov.justice.digital.config.JobParameters;
+import uk.gov.justice.digital.domain.model.SourceReference;
+import uk.gov.justice.digital.service.DataStorageService;
+import uk.gov.justice.digital.service.SourceReferenceService;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import static org.apache.spark.sql.types.DataTypes.StringType;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.*;
+
+public class CuratedZoneTest extends BaseSparkTest {
+
+    private static final String S3_PATH_KEY = "dpr.curated.s3.path";
+    private static final String S3_PATH = "s3://loadjob/curated";
+
+    private final JobParameters jobParameters = new JobParameters(Collections.singletonMap(S3_PATH_KEY, S3_PATH));
+    private final DataStorageService storage = new DataStorageService();
+
+    @Mock
+    private Dataset<Row> mockedDataSet;
+
+    @Mock
+    private Row mockedRow;
+
+
+    @BeforeEach
+    void setUp() {
+    }
+
+    @Test
+    public void shouldReturnValidCuratedS3Path() {
+        val source = "oms_owner";
+        val table  = "agency_internal_locations";
+        val operation = "load";
+        val expectedCuratedS3Path = String.join("/", S3_PATH, source, table, operation);
+        assertEquals(expectedCuratedS3Path, this.storage.getTablePath(S3_PATH, source, table, operation));
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
```

Comment:
Apply this to the class instead of the method so if additional cases are added the mocks are populated.